airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -328,11 +328,16 @@ def download(
         # TODO: future improvement check file size before downloading,
         #  to check for local space availability
 
-        num_file_attempts = 0
+        if num_max_attempts is None:
+            num_max_attempts = 3
+
+        for num_file_attempts in range(num_max_attempts):
```

Comment:
Probably would help reduce the changeset if this is `range(1, num_max_attempts + 1)`. A bit more verbose but easier to understand.

Reply:
`num_max_attempts` can also be `None` and then the test fails.

Reply:
Isn’t it set to 3 in that case?